cmd/samples/recipes/cancelactivity/workflow.go

```diff
@@ -60,7 +73,9 @@ func activityToBeCanceled(ctx context.Context) (string, error) {
 			activity.RecordHeartbeat(ctx, "")
 		case <-ctx.Done():
 			logger.Info("context is cancelled")
-			return "I am canceled by Done", nil
+			// returned canceled error here so that in workflow history we can see ActivityTaskCanceled event
+			// or if not cancelled, return timeout error
+			return "I am canceled by Done", ctx.Err()
```

Comment:
Only error will be accessible in the workflow, and I think if there's an error the return result isn't recorded at all.  Since errors are modeled as exceptions in java (and probably other langs), that's probably a good thing, since it's impossible to get both an exception and a return value there.